ports/esp32/modsocket.c

```diff
@@ -942,6 +942,8 @@ static mp_obj_t esp_socket_getaddrinfo(size_t n_args, const mp_obj_t *args) {
     if (n_args > 5) {
         hints.ai_flags = mp_obj_get_int(args[5]);
     }
+    // Since we rely on ai_canonname after this call (in _getaddrinfo_inner), we explicitly ask for it
+    hints.ai_flags = hints.ai_flags | AI_CANONNAME;
```

Comment:
There are two calls to `_getaddrinfo_inner()`.  Does the other one also need to set this hint?  Because `_getaddrinfo_inner()` accesses the `ai_canonname` entry.

Reply:
if add for other, works.
static void _socket_getaddrinfo(const mp_obj_t addrtuple, struct addrinfo **resp) {
    mp_obj_t *elem;
    mp_obj_get_array_fixed_n(addrtuple, 2, &elem);
    struct addrinfo hints = { 0 };
    // Add AI_CANONNAME flag
    hints.ai_flags = AI_CANONNAME;
    
    _getaddrinfo_inner(elem[0], elem[1], &hints, resp);
}